### PR TITLE
Add fxhash_32

### PR DESCRIPTION
**Pull request by sffc**
Utility function prerequisite for #1148

### CODE REVIEW

**Comment by sffc** [comment]
@Manishearth @hsivonen @alolita PTAL at the boilerplate I added and verify that I am attributing the code correctly.

**Review by Manishearth — APPROVED** [review]
good from my end :+1: 

might be worth asking rustc_hash to make their code const

**Comment by sffc** [comment]
^ this is a squash.

**Comment by sffc** [comment]
^ this is a rebase.

**Review by Manishearth — APPROVED** [review]

**Comment by sffc** [comment]
^ rebase again.

Waiting for approval from @hsivonen or @alolita

File: provider/core/src/helpers.rs [hunk 1]
```diff
@@ -76,3 +76,70 @@ fn test_escape_for_json() {
         escape_for_json("ab\u{001F}c", &mut String::new())
     );
 }
+
+/// Const function to compute the FxHash of a byte array with little-endian byte order.
+#[allow(dead_code)]
+pub const fn fxhash_32(bytes: &[u8]) -> u32 {
```
Comment (robertbastian) [inline provider/core/src/helpers.rs:82]: Please add a comment why you chose this hash function.

Comment (sffc) [reply provider/core/src/helpers.rs:96]: Done.

**Review by robertbastian — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]

**Comment by sffc** [comment]
In order to get unblocked, I am going to merge this PR.  I am still going to be on the lookout for a review from @alolita and make a follow-up PR with any necessary changes.

**Pull request merged**